### PR TITLE
[native] Add e2e test for aggregation companion function

### PR DESCRIPTION
**Pull request by kagamiori**
Test plan

Add an e2e test for the companion functions of the `avg()` aggregation function in `testAggregationCompanionFunction`. The test case compares the result of a query using avg companion function against a semantically equivalent query using the Presto avg function. This test case is added to assert the correctness of auto-generated companion functions.

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-native-execution/src/test/resources/external_functions.json
Comment (mbasmanova) [inline presto-native-execution/src/test/resources/external_functions.json:40]: We should not overwrite existing Presto functions. Let's remove this entry.

File: presto-native-execution/src/test/java/com/facebook/presto/spark/TestPrestoSparkNativeSimpleQueries.java
Comment (mbasmanova) [inline presto-native-execution/src/test/java/com/facebook/presto/spark/TestPrestoSparkNativeSimpleQueries.java:124]: These queries do not reflect the way companion functions are designed to be used. Would it be possible to create more realistic tests by (1) evaluating 'partial' aggregation and save results to a table; (2) evaluate 'final' aggregation on results of (1).

File: presto-native-execution/src/test/resources/external_functions.json
Comment (kagamiori) [reply presto-native-execution/src/test/resources/external_functions.json:40]: Thanks for the review! I thought all functions needs to be registered through this JSON file. Let me double check with @Ali-P.

Comment (Ali-P) [reply presto-native-execution/src/test/resources/external_functions.json:40]: - This UDF is accessible only by `json.test_schema.avg(...)` as of now (so, it doesn't overwrite `presto.default.avg(...)`)
- You need to register the functions you use in the queries. I don't see you using `json.test_schema.avg` UDF in the queries. So, you can remove `avg` entry from the json file UNLESS you've listed it as **Documentation** (which would make sense to keep the registration).

**Comment by kagamiori** [comment]
Blocked by https://github.com/facebookincubator/velox/pull/5633. The failure of linux-spark-e2e-tests will be fixed after that gets landed.

File: presto-native-execution/src/test/java/com/facebook/presto/spark/TestPrestoSparkNativeSimpleQueries.java
Comment (Ali-P) [inline presto-native-execution/src/test/java/com/facebook/presto/spark/TestPrestoSparkNativeSimpleQueries.java:132]: The updated test seems to be representative of the feature's application. 

nit: If companion function is a *niche* feature, I recommend adding a doc-str (comment) on what the test is doing; otherwise, ignore this comment. :)

File: presto-native-execution/src/test/resources/external_functions.json
Comment (Ali-P) [inline presto-native-execution/src/test/resources/external_functions.json:109]: The json entries LGTM; 
nit: it may be beneficial to use a more specific naming; for e.g. instead of `avg_merge`, use `avg_merge_only` / `avg_merge_intermediate` to indicate this is not a final step (AOT regular agg functions where there's only one reduction / merge step). one can definitely infer this from the test cases; so, please ignore if you think the current naming is clear.

File: presto-native-execution/src/test/java/com/facebook/presto/spark/TestPrestoSparkNativeSimpleQueries.java
Comment (Ali-P) [inline presto-native-execution/src/test/java/com/facebook/presto/spark/TestPrestoSparkNativeSimpleQueries.java:161]: If the assertion fails (or any exception is raised before `DROP TABLE` instruction, the table will not be dropped; 
if this is acceptable, no changes is required, otherwise, I would use try-finally to make sure the table is dropped at tall times.

File: presto-native-execution/src/test/resources/external_functions.json
Comment (kagamiori) [reply presto-native-execution/src/test/resources/external_functions.json:109]: Hi @Ali-P, the companion function names (e.g., avg_merge) are auto-generated and follow a name convention (https://github.com/facebookincubator/velox/discussions/4493). E.g, for avg(), its companion functions are avg_partial, avg_merge, etc. For another aggregation function sum(), its companion functions are sum_partial, sum_merge, etc. So I cannot just modify the name of avg's companion functions.

Comment (Ali-P) [reply presto-native-execution/src/test/resources/external_functions.json:109]: SGTM :)

File: presto-native-execution/src/test/java/com/facebook/presto/spark/TestPrestoSparkNativeSimpleQueries.java
Comment (Ali-P) [inline presto-native-execution/src/test/java/com/facebook/presto/spark/TestPrestoSparkNativeSimpleQueries.java:150]: nit: `CREATE TABLE` is better off outside of try-finally since it's failure does not need the table to be dropped; this follows the principle of minimizing try-blocks' scope.

Comment (kagamiori) [reply presto-native-execution/src/test/java/com/facebook/presto/spark/TestPrestoSparkNativeSimpleQueries.java:150]: Hi @Ali-P, I saw that test cases in AbstractTestNativeGeneralQueries.java put CREATE TABLE inside the try-blocks, so I followed that. I don't have a strong opinion. Let me know what you think.

Comment (Ali-P) [reply presto-native-execution/src/test/java/com/facebook/presto/spark/TestPrestoSparkNativeSimpleQueries.java:150]: I see; leaving it outside of try-catch is better (one would need a reason for including a statement in the try-block, which there is none here). But the PR is GTG (as the PR is similar to other examples). No need to update. :)

**Review by mbasmanova — APPROVED** [review]

**Pull request merged**
